src/esp/bindings/SimBindings.cpp

```diff
@@ -74,6 +74,8 @@ void initSimBindings(py::module& m) {
 
   // ==== Simulator ====
   py::class_<Simulator, Simulator::ptr>(m, "Simulator")
+      // add constructor to build Simulator from existing MetadataMediator
+      .def(py::init<esp::metadata::MetadataMediator::ptr>())
```

Comment:
Mind adding a pytest test case somewhere. It will serve as documentation too.

Reply:
working on it now